airbyte-integrations/csv/src/main/java/io/airbyte/integrations/csv/CsvDestination.java

```diff
@@ -0,0 +1,190 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.csv;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Preconditions;
+import com.google.common.io.Resources;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.DestinationConnectionSpecification;
+import io.airbyte.config.Schema;
+import io.airbyte.config.StandardCheckConnectionOutput;
+import io.airbyte.config.StandardCheckConnectionOutput.Status;
+import io.airbyte.config.StandardDiscoverSchemaOutput;
+import io.airbyte.config.Stream;
+import io.airbyte.integrations.javabase.Destination;
+import io.airbyte.integrations.javabase.DestinationConsumer;
+import io.airbyte.integrations.javabase.JavaBaseConstants;
+import io.airbyte.singer.SingerMessage;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CsvDestination implements Destination {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CsvDestination.class);
+
+  private static final String DESTINATION_PATH_FIELD = "destination_path";
+
+  // todo (cgardens) - hack to handle the fact that the resources of this jar aren't in original
+  // classpath. should try to find a way to handle this
+  // outside of the runtime.
+  private URLClassLoader getClassLoader() {
+    final String destinationJarPath = System.getenv().get(JavaBaseConstants.ENV_DESTINATION_JAR_PATH);
+
+    try {
+      return new URLClassLoader(new URL[] {Path.of(destinationJarPath).toUri().toURL()});
+    } catch (MalformedURLException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @SuppressWarnings("UnstableApiUsage")
+  @Override
+  public DestinationConnectionSpecification spec() throws IOException {
+    final String resourceString = Resources.toString(getClassLoader().findResource("spec.json"), StandardCharsets.UTF_8);
+    return Jsons.deserialize(resourceString, DestinationConnectionSpecification.class);
+  }
+
+  @Override
+  public StandardCheckConnectionOutput check(JsonNode config) {
```

Comment:
we could use json2pojo to generate a type so that we do not need to use raw JsonNode. on the fence as to whether it is worth it.